### PR TITLE
fix(server): timeout in server creation when waiting on next actions

### PR DESCRIPTION
**Pull request by mmontesi**
##### SUMMARY
Timeout in server creation is insufficient when using large images (like Rocky8/9).
Random errors depending on image size and location load (server creation can take up to 5 minutes and more).

Similar to this one: https://github.com/ansible-collections/hetzner.hcloud/pull/189

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
module: server

##### ADDITIONAL INFORMATION
Error:
```
fatal: [test-master -> localhost]: FAILED! => {"changed": false, "failure": {"action": {"command": "start_server", "error": null, "finished": null, "id": 1772219884, "progress": 0, "resources": [{"id": 52912003, "type": "server"}], "started": "2024-09-10T09:30:35+00:00", "status": "running"}}, "msg": "The pending action timed out"}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible_collections.hetzner.hcloud.plugins.module_utils.vendor.hcloud.actions.domain.ActionTimeoutException: The pending action timed out
```
Tested after the changes with cpx51 and rocky-9, no more errors.

### CODE REVIEW

**Comment by jooola** [comment]
The current timeout is already ~30m long, I am hesitant to increase this even more.

1000 retries, is way too long, to approximate the timeout duration you can do max_retries * 5 = timeout in seconds. So we would have a timeout of ~1h22 minutes.

Could you maybe provide some more information so I can try to reproduce this and see how long it should take ?

**Comment by apricote** [comment]
I think the issue is that the action `start_server` is not the one with the long timeout. The primary action being returned is `create_server`, but the `next_actions` use the default wait time which is at around 120 seconds right now.

**Comment by mmontesi** [comment]
> I think the issue is that the action `start_server` is not the one with the long timeout. The primary action being returned is `create_server`, but the `next_actions` use the default wait time which is at around 120 seconds right now.

Yes, you're right. I wasn't able to deploy until I increased the second timeout. We could probably use the same value for both.
As soon as I come back from vacation, I'll do some more tests ;)
I noticed that after the server is created (cpu + memory allocated), it can sometimes take 15-20 minutes to connect it to the network, `max_retries=362` should be enough.

File: plugins/modules/server.py [hunk 1]
```diff
@@ -471,9 +471,9 @@ def _create_server(self):
                 self.result["root_password"] = resp.root_password
                 # Action should take 60 to 90 seconds on average, but can be >10m when creating a
                 # server from a custom images
-                resp.action.wait_until_finished(max_retries=362)  # 362 retries >= 1802 seconds
+                resp.action.wait_until_finished(max_retries=1000)  # 362 retries >= 1802 seconds
```
Comment (apricote) [inline plugins/modules/server.py:474]:
```suggestion
                resp.action.wait_until_finished(max_retries=362)  # 362 retries >= 1802 seconds
```

As far as I could tell the issue happens not with the `create_server` action that is being waited on here, but with the `resp.next_actions` 2 lines below.

File: plugins/modules/server.py [hunk 2]
```diff
@@ -471,9 +471,9 @@ def _create_server(self):
...
                 for action in resp.next_actions:
-                    action.wait_until_finished()
+                    action.wait_until_finished(max_retries=1000)
```
Comment (apricote) [inline plugins/modules/server.py:476]:
```suggestion
                    # Starting the server or attaching to the network might take a few minutes,
                    # depending on the current activity in the project.
                    # This waits up to 30minutes for each action in series, but in the background
                    # the actions are mostly running in parallel, so after the first one the other
                    # actions are usually completed already.
                    action.wait_until_finished(max_retries=362) # 362 retries >= 1802 seconds
```

Please change this to a reasonable and documented amount of retries/wait time.

**Review by apricote — CHANGES_REQUESTED** [review]
In addition to these two changes, could you please add a changelog fragment, similar to https://github.com/ansible-collections/hetzner.hcloud/commit/49e21ae4febe18c62e44c9abf365fa2feee7e7cc#diff-37f5b6e5915687ad0c810d176e446e179b1c8a5a8193245319508eabcc9b8dc6 in this PR?

```yaml
bugfixes:
  - server - Wait up to 30 minutes for every action returned from server create
```

**Comment by apricote** [comment]
We had some more customers that reported this issue, so I went ahead and applied the changes in #561. Thanks for opening the PR!

**Pull request closed without merging**